### ISSUE TITLE
"Match Case" & "Regex" difficult to see in Searchbar 

### ISSUE THREAD

**Issue by newsgrep** [lanoxx/tilda#220]
pic: ![tilda_bug2](https://cloud.githubusercontent.com/assets/13681304/12529887/c60e8678-c1c9-11e5-921c-1328e341273d.png)

when I exit Tilda I get:

(tilda:8279): GLib-GObject-WARNING **: invalid unclassed pointer in cast to 'GtkNotebook'

(tilda:8279): Gtk-CRITICAL **: gtk_notebook_get_n_pages: assertion 'GTK_IS_NOTEBOOK (notebook)' failed

running Debian Testing 64Bit + Tilda 1.3.1 + i3wm 4.11

**Comment by lanoxx** [comment]
The visual problem looks like a theme problem to me. Are you using the Adwaita theme from Gtk+? On my system the background of the searchbar is white or some light grey. 

The errors look like there is a problem in the code, i will take a look.

**Comment by newsgrep** [comment]
I use the Raleigh Theme. The problem does happen with the Adwaita Theme to. I think there was an 1.3-Alpha Version were it worked just fine (when I remember correctly).

**Comment by lanoxx** [comment]
I just tried to reproduce the warnings on GtkNotebook, but I could not get it to happen on my system. 

1.) Can you reproduce this with the default settings?
2.) If not, do these warnings only appear when using certain settings?
3.) Regarding the style problem, do you use a custom ~/.config/tilda/style.css file where you overwrite some background?
4.) Could you try to bisect with git and find the version where the background started to become broken for you?

**Comment by newsgrep** [comment]
The GtkNotebook warning is triggered when I close the tilda Window with ALT+F4 (i3 command "kill") , it does not happen when I Quit from the context menu. 
1. yes
3. I do not use a style.css
4. I will have a look into, but it is new to me.

**Comment by newsgrep** [comment]
setting  gtk_widget_set_app_paintable (GTK_WIDGET (tw->window), <b>FALSE</b>); 
fixes this bug.

**Comment by lanoxx** [comment]
I cannot remove that. Its needed in gnome shell.

**Comment by lanoxx** [comment]
could you try to set only the searchbar widget'S app paintable flag to false?

**Comment by newsgrep** [comment]
I'm sorry, I should do what?

**Comment by lanoxx** [comment]
Try adding:

`gtk_widget_set_app_paintable (GTK_WIDGET (tw->search->search_box), FALSE);`

And leave the other line at true.

**Comment by newsgrep** [comment]
file: src/tilda_window.c 
 941     gtk_widget_set_app_paintable (GTK_WIDGET (tw->window), TRUE);
 942     gtk_widget_set_app_paintable (GTK_WIDGET (tw->search->search_box), FALSE);  

turns out to be a segfault

**Comment by lanoxx** [comment]
I pushed a new commit, that sets a widget name for the search bar. Was a work around you can try to add a style.css with the following rule:

    #search {
        background: #fff;
    }

Let me know if that fixes it.

**Comment by newsgrep** [comment]
your new code + the style.css file fixes it.

**Comment by lanoxx** [comment]
I pushed another commit to master that adds default styling to tilda. It should now also work without adding custom styling to style.css.

I have also committed some patch to fix some deprecation warnings and casting warnings when compiling tilda, the output should be more clean now.

Regarding the warnings when you close tilda with `ALT+F4`, it seems that's caused by a call to `tilda_window_free(&tw);` after the gtk_main loop has already closed and so the resources are already freed. Fixing this properly will require a bit more work so I will not do it right now, feel free to open separate bug report for it, so it does not get lost when I close this one.
